frontend/packages/console-shared/src/hooks/useBoundingClientRect.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { useResizeObserver } from './useResizeObserver';
+
+type BoundingClientRect = ClientRect | null;
+
+export const useBoundingClientRect = (targetElement: HTMLElement | null): BoundingClientRect => {
+  const [clientRect, setClientRect] = React.useState<BoundingClientRect>(null);
+
+  const observerCallback = React.useCallback(() => {
+    setClientRect(targetElement ? targetElement.getBoundingClientRect() : null);
+  }, [targetElement]);
+
+  React.useEffect(() => {
+    if (targetElement) {
+      setClientRect(targetElement.getBoundingClientRect());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why do you need to disable the es lint rule here? Target element shouldn't get changed anyway.

Reply:
Only wanted this to run when the component is mounted.

Reply:
Use `useState(function)` for lazy initialization